src/content/reference/react/use-server.md

```diff
@@ -24,34 +24,233 @@ canary: true
+
+<Note>
+
+We force `UsernameForm` as a Client Component with `'use client'` purely for demonstration. In general, it is not recommended to add the `'use client'` directive if the component does not use [client-only features](reference/react/use-client#usage).
+
+</Note>
+
+#### Progressive Enhancement in `<form>` {/*progressive-enhancement-in-form*/}
+
+In the above example, the `<form>` is rendered in a Client Component. By instead rendering the `<form>` in a Server Component, React can support [progressively enhanced](https://developer.mozilla.org/en-US/docs/Glossary/Progressive_Enhancement) forms.
+
+In this context, we mean that server-rendered forms can provide the baseline experience of submitting a form without JavaScript. This is useful for apps that are server-side rendered (SSR) where HTML scaffolding is sent to the client first, before any code is downloaded to add interactivity.
+
+<DeepDive>
+
+#### What is the difference between SSR and React Server Components (RSC)? {/*what-is-the-difference-between-ssr-and-react-server-components-rsc*/}
+
+Server-side rendering and React Server Components are two features for optimizing performance of React apps and can complement one another.
+
+Server-side rendering (SSR) is the concept of taking a single render pass of a React app to resolve it to its base UI primitives, HTML elements for the web. This HTML is sent down to the client to serve as "scaffolding" while JavaScript is still being downloaded and evaluated.
+
+SRR is an optimization for ["time to paint"](https://developer.mozilla.org/en-US/docs/Glossary/First_paint) but it still requires all your app code to download and be evaluated on the client. The client then needs to [hydrate](/reference/react-dom/client/hydrateRoot#hydrating-server-rendered-html) this scaffolding to add interactivity.
+
+React Server Components (RSC) is the concept of breaking up your React [module dependency tree](/reference/react/use-client#marking-client-components-with-use-client) to determine what components can be rendered on the server and what can be sent to the client. In contrast to SSR, components that are rendered on the server have access to server APIs and can make data-fetches.
+
+These rendered Server Components are streamed down to the client in a special encoding to be translated to HTML or whatever client UI primitive. There is no hydration due to [limitations of Server Components](https://react-dev-git-use-client-fbopensource.vercel.app/reference/react/use-client#limitations).
+
+RSC is an optimization for ["time to interactive"](https://developer.mozilla.org/en-US/docs/Glossary/Time_to_interactive) as it reduces the code that needs to be downloaded and run by the client.
+
+</DeepDive>
+
+Here is the same username request form but rendered as a Server Component.
```

Comment:
would it make sense to just remove the "use client" without combining the files? just mention that they can be combined

Reply:
I think I'll remove the separation of these examples -- I wanted to keep them separate as I thought there was a difference between server / client rendering the form but I was wrong. I mean yes `useFormState` can only be used in client component, but I don't think it's worth focussing on render environments beyond that.